client/src/manage/Substack.tsx

```diff
@@ -0,0 +1,251 @@
+import React, { useEffect, useState } from 'react'
+import { FlexColumn, Main, Row } from '../components/Layout'
+import { BaseText, Desc, DescLeft, SmallText, Title } from '../components/Text'
+import config from '../../config'
+import { Button, Input, LinkWrarpper } from '../components/Controls'
+import { buildClient, EWSTypes } from '../api'
+import { getSld, getSubdomain } from '../utils'
+import { isValidSubstackLandingUrl } from '../../../common/substack-utils'
+import { Feedback, Loading } from '../components/Misc'
+import useDebounce from '../hooks/useDebounce'
+import styled from 'styled-components'
+import { toast } from 'react-toastify'
+import { useTryCatch } from '../hooks/useTryCatch'
+import { ethers } from 'ethers'
+import detectEthereumProvider from '@metamask/detect-provider'
+import { type ExternalProvider } from '@ethersproject/providers'
+import { EthereumProvider } from '@walletconnect/ethereum-provider'
+
+const Container = styled(Main)`
```

Comment:
We can put common components and connect-wallet hooks and functions in a common file, later